src/app.provider.ts

```diff
@@ -96,7 +95,7 @@ export class TestAppProvider extends AppProvider {
   }
 
   protected getApp(module: any): Promise<INestApplication> {
-    if (!(module instanceof TestingModule))
+    if (module.constructor.name !== 'TestingModule')
```

Comment:
I took this approach because `TestingModule` belongs to `@nestjs/testing`, which is not installed as a production dependency. I thought this was the simpler way to remove `TestingModule` import from this file, but alternatives like including `@nestjs/testing` as a production dependency, or even moving `TestAppProvider` class to a separate file under a `__tests__` folder (so it would get ignored when building the production image), could be also valid approaches.

As I said I think I took the simpler alternative but I'm open to suggestions.

Reply:
In my opinion I'd do something like it was done [here](https://github.com/safe-global/safe-gelato-relay-service/pull/99).

I'd say that it can be a separate PR on which this depends on. Isolating testing logic shouldn't be directly linked to Cache/Docker.

Reply:
> I'd say that it can be a separate PR on which this depends on.

I agree, I will tackle this in a separate PR, and leave this one as a draft.

Reply:
Done https://github.com/safe-global/safe-client-gateway-nest/pull/526 🙂